lib/utils/isValidIdentifier.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+/**
+ * returns whether a string is a valid CSS identifier
+ * (i.e. only alphanumeric characters, `-`, and `_`;
+ * does not have a leading digit, leading dash followed by digit, or two leading dashes)
+ * furthermore, any escaped or ISO 10646 characters are allowed.
+ * see: https://www.w3.org/TR/CSS2/syndata.html#value-def-identifier
+ * @param {string} ident
+ * @returns {boolean}
+ */
+module.exports = function isValidIdentifier(ident) {
```

Comment:
question

I dunno the stylelint conventions, but did you check if a npm already existed?
Even if it does maybe we recommend not relying on dependencies if possible.

Reply:
Hm, to be honest I didn't look too in-depth; I did a quick `site:npmjs.com valid css identifier` and found nothing on the first page or so of Google (much more support for valid JS identifiers or CSS selectors). If you're aware of one and we think we should use it though, happy to refactor!